src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
+type Color = 'red' | 'blue' | 'green';
 export interface Figure {
-
+  shape: 'circle'| 'triangle' | 'rectangle';
+  color: Color;
+  getArea():number;
 }
 
 export class Triangle implements Figure {
+  public shape: 'triangle';
+
+  constructor(
+    public color:Color,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    this.shape = 'triangle';
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('You can\'t make a triangle with negative number(s)');
+    }
+
+    if ((this.a + this.b) <= this.c) {
```

Comment:
I think any side can be the longest, not only this.c, so you need to check all possiple cases

Reply:
right remark